EGO Forum Enhancement.user.js

```diff
@@ -0,0 +1,301 @@
+    subforum.innerHTML = "<div class=\"node-body\"><span class=\"node-icon\"><i class=\"fa--xf far fa-comments\" aria-hidden=\"true\"></i></span><div class=\"node-main js-nodeMain\"><h3 class=\"node-title\"><a href=\"/forums/685/\" data-xf-init=\"element-tooltip\" data-shortcut=\"node-description\">Moderator Trash Bin</a></h3></div><div class=\"node-extra\"><span class=\"node-extra-placeholder\">Planes, Trains, and Plantains</span></div></div>";
+    private_category.appendChild(subforum);
+}
+
+/**
+ * Handles generic/nonspecific threads
+ */
+function handleGenericThread() {
+    var breadcrumbs = document.querySelector(".p-breadcrumbs").innerText;
+    if (breadcrumbs.match(/((Contest (a Ban|Completed))|(Report (a Player|Completed))) ?$/)) { // Ban Contest or Report
+        handleBanReport();
+    }
+    if (isLeadership(breadcrumbs)) // LE Forums
+        handleLeadership();
+}
+
+/**
+ * Adds "View Bans" or "Lookup ID" button on report/contest threads.
+ * TODO: Add support for other game IDs
+ */
+function handleBanReport() {
+    var breadcrumbs = document.querySelector(".p-breadcrumbs").innerText;
+    var post_title = document.querySelector(".p-title").innerText;
+    var button_group = document.querySelector("div.buttonGroup");
+    addMAULProfileButton(button_group, document.querySelector(".message-name > a.username").href.substring(35));
+
+    var steam_id = post_title.match(/^.* - .* - ([^\d]*?(?<game_id>(\d+)|(STEAM_\d:\d:\d+)|(\[U:\d:\d+\])).*)$/)
+    if (steam_id) {
+        var unparsed_id = steam_id.groups.game_id;
+        try {
+            var steam_id_64 = (SteamIDConverter.isSteamID64(unparsed_id) ? unparsed_id : SteamIDConverter.toSteamID64(unparsed_id));
+            addBansButton(button_group, steam_id_64);
+        } catch (TypeError) {
+            addBansButton(button_group, post_title.split(" - ")[2]);
+            addLookupButton(button_group, post_title);
+        }
+    } else {
+        addBansButton(button_group, post_title.split(" - ")[2]);
+        addLookupButton(button_group, post_title);
+    }
+
+    if (!breadcrumbs.match(/Completed ?$/))
+        addMoveButton(button_group, window.location.href);
+}
+
+/**
+ * Adds Confidential banners on top and bottom of page
+ */
+function handleLeadership() {
+    generateRedText("5%");
+    generateRedText("80%");
+}
+
+/**
+ * Generates large, transparent text (basically a watermark)
+ * @param {string} top CSS Top Style
+ * @param {string} str Text to display
+ */
+function generateRedText(top, str = "Confidential") {
+    var text = document.createElement("div");
+    document.body.appendChild(text);
+
+    text.innerHTML = str;
+    text.style.color = "rgba(255,0,0,0.25)";
+    text.style.fontSize = "100px";
+    text.style.position = "fixed";
+    text.style.top = top;
+    text.style.left = "50%";
+    text.style.transform = "translateX(-50%)"
+    text.style.pointerEvents = "none";
+    text.style.zIndex = "999";
+}
```

Comment:
Would be ideal to move this up to near the top of the file where all the other helper functions are. The file follows a nice layout right now, although this particular function is an outlier.